evm/src/cpu/kernel/asm/memory/core.asm

```diff
@@ -1,3 +1,223 @@
+// Load a big-endian u32, consisting of 4 bytes (c_3, c_2, c_1, c_0).
+%macro mload_u32
```

Comment:
Ah true — although this should be _slightly_ more efficient because it doesn't have to check the length after each byte. But using `mload_packing` would be cleaner, and reduce code size (since it's a function instead of a macro) so do you think that'd be preferable?

Reply:
Yeah true the length checking is a bit redundant in this context, but it shouldn't really matter compared to the overall costs of the precompiles. 
More important is to have these be functions since I think they're a bit too large to be macros (especially `mload_u256`).

Reply:
Ah okay! I'll just use mload_packing and mstore_packing then